File: src/ros/Utils/RosbagWriter.jl

export RosbagWriter

# assumed this path was added as part of the bag subscriber code
# pushfirst!(PyVector(pyimport("sys")."path"), @__DIR__ )

writeRosbagPy = pyimport("rosbagWriter")
RosbagWriter = writeRosbagPy."RosbagWriter"


# Base.@kwdef struct RosbagWriter
#   bagfile::String
#   bag::PyObject = writeRosbag(bagfile)
# end

# function write(rbw::RosbagWriter, channel::AbstractString, msg)
#   rbw.bag.write_message(channel, msg)
# end

# function close()

File: src/ros/Utils/rosbagWriter.py

import rosbag
import sys

class RosbagWriter:
    def __init__(self, bag_file_name, topic_name):
        #bag_file_name = sys.argv[1]
        #topic_name = sys.argv[2]

        self.bag = rosbag.Bag(bag_file_name)
        self.bag_contents = self.bag.read_messages(topics=[topic_name])

        self.idx = 0
        self.topic_size = self.bag.get_message_count(topic_filters=topic_name)

    def write_message(self,channel,msg,header=msg.header.stamp):
        if (self.idx < self.topic_size):
            self.bag.write(channel, msg, header)
        else:
            self.bag.close()
            return False